### DISCUSSION TITLE
커머스솔루션마켓 플러그인을 개발하면서 플러그인 설치 스토어의 정보는 어떻게 얻을 수 있나요?

### DISCUSSION THREAD

**Discussion by Philip-general** [commerce-api-naver/commerce-solutions#48] [묻고 답하기]
스토어 주문정보와 같이 commerce-api에서 호출하여 얻을 수 있는 정보들을 플러그인 개발사에서 호출할 수 있나요?
권한 부여 단계 같은 것이 존재하는지 궁금합니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

> [!IMPORTANT]
> 게시해주신 내용은 커머스솔루션 개발사 연동 관련 문의로 판단되어 관리자가 게시 공간을 이동하였습니다.
> - 최초 게시된 공간: 커머스API 내 스토어 애플리케이션 기술지원 공간 (`/commerce-api`)
> - 이동된 공간: 커머스솔루션 개발사 기술지원 공간 (`/commerce-solutions`)

커머스솔루션 가입/사용 과정에서 솔루션 시스템에 전달하는 암호화 된 토큰을 이용하여 대상 스토어 정보를 확인할 수 있습니다.

- 기본 설정은 'JWT 토큰' 전달로 솔루션 구독자가 커머스솔루션마켓 혹은 스마트스토어센터 솔루션 목록에서 [사용하기] 버튼을 선택하여 솔루션 웹 사이트로 이동 시 URL 파라미터 `token`으로 JWT 토큰을 전달합니다.
이 JWT 토큰을 솔루션 시스템만 알고있는 '솔루션 공개키'를 이용하여 복호화를 하면 대상 구독자의 구독 정보를 확인할 수 있습니다.
    - 상세 내용: [커머스API센터 > 솔루션 개발 가이드 > 기본 연동 요소 가이드](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%EA%B8%B0%EB%B3%B8-%EC%97%B0%EB%8F%99-%EC%9A%94%EC%86%8C-%EA%B0%80%EC%9D%B4%EB%93%9C)

- '심사 후 승인' 형 설정을 활성화한 경우 사용자의 구독 신청 후 '판매자 커머스ID 인증'을 통해 확인이 가능합니다.
'판매자 커머스ID 인증'이 완료되면 지정된 솔루션 웹 사이트 URL(Redirect URL)로 JWE 토큰이 전달됩니다. 전달된 JWE 토큰은 솔루션 시스템 로컬에서 복호화가 불가능하며 전용 API로 복호화가 가능합니다.
    - 관련 API: [솔루션] 판매자 인증 JWE 해석 (`GET /v1/commerce-solutions/seller-info-by-token`)
    - 상세 내용: [커머스API센터 > 솔루션 개발 가이드 > 판매자 커머스 아이디 인증](https://apicenter.commerce.naver.com/ko/basic/solution-doc#section/%EA%B0%9C%EB%B0%9C%ED%95%98%EA%B8%B0/%ED%8C%90%EB%A7%A4%EC%9E%90-%EC%BB%A4%EB%A8%B8%EC%8A%A4-%EC%95%84%EC%9D%B4%EB%94%94-%EC%9D%B8%EC%A6%9D)

**Reply by Philip-general** [reply]
![image](https://github.com/user-attachments/assets/f7be4e27-b41d-4477-876b-d160dec8a099)
친절한 답변 감사드립니다.
확인차 한 번 더 질문을 드립니다

(심사 후 승인형 전용) 의 플로우에 셀러가 판매자 정보 조회 허용을 하면 SELLER 인증 토큰을 받을 수 있게 되고,
해당 토큰으로 commerce-api의 사용이 가능한 것이라고 이해를 해도 괜찮을까요?

**Reply by commerce-api-naver** [reply]
@Philip-general 님 안녕하세요. 커머스API 이민석입니다.

네, 맞습니다.
정확하게는 아래와 같은 순서 후 대상 구독자(판매자)에 대한 커머스API 호출이 가능합니다.

1. [사용자] 사용자(판매자)의 커머스솔루션 구독 신청
2. [사용자] "1번" 과정 후 사용자(판매자)의 대상 솔루션 웹 사이트 방문
    - 솔루션 웹 사이트 내에서의 회원가입 또는 사용자 로그인 등이 필요할 경우 수행
3. [사용자] 사용자(판매자)의 '판매자 커머스ID 인증' 수행
4. [솔루션] 전달받은 JWE 토큰 값을 이용한 내부 시스템 처리
    - 솔루션 자체 회원 정보와 JWE 토큰 내 구독 신청 정보 처리 등
5. **[솔루션] '사용 시작 승인' API (`PUT /v1/commerce-solutions/subscriptions/approve`)' 호출**

'심사 후 승인'형 기능은 커머스솔루션마켓에서 구독자의 솔루션 구독 신청과는 별개로 솔루션 시스템에서의 회원 인증(회원 정보 연결), 구독 신청 정보 검토 후 구독 승인 등 별도 절차가 필요할 경우 사용할 수 있습니다.
